packages/admin-next/dashboard/src/routes/orders/order-create-fulfillment/components/order-create-fulfillment-form/order-create-fulfillment-form.tsx

```diff
@@ -0,0 +1,192 @@
+import { Form } from "../../../../../components/common/form"
+import { OrderCreateFulfillmentItem } from "./order-create-fulfillment-item"
+import { getFulfillableQuantity } from "../../../../../lib/line-item"
+
+type OrderCreateFulfillmentFormProps = {
+  order: Order
+}
+
+/**
+ * TODO: support fulfilllments for Claims and Swaps
+ */
+
+export function OrderCreateFulfillmentForm({
+  order,
+}: OrderCreateFulfillmentFormProps) {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+
+  const { mutateAsync: createOrderFulfillment, isLoading: isMutating } =
+    useAdminCreateFulfillment(order.id)
+
+  const [fulfillableItems, setFulfillableItems] = useState(() =>
+    order.items.filter((item) => getFulfillableQuantity(item) > 0)
+  )
+
+  const form = useForm<zod.infer<typeof CreateFulfillmentSchema>>({
+    defaultValues: {
+      quantity: fulfillableItems.reduce((acc, item) => {
+        acc[item.id] = getFulfillableQuantity(item)
+        return acc
+      }, {} as Record<string, number>),
+      send_notification: !order.no_notification,
+    },
+    resolver: zodResolver(CreateFulfillmentSchema),
+  })
+
+  const { stock_locations = [] } = useAdminStockLocations({})
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    await createOrderFulfillment({
+      location_id: data.location_id,
+      no_notification: !data.send_notification,
+      items: Object.entries(data.quantity)
+        .filter(([, value]) => !!value)
+        .map(([item_id, quantity]) => ({ item_id, quantity })),
+    })
+
+    handleSuccess(`/orders/${order.id}`)
+  })
+
+  const onItemRemove = (itemId: string) => {
+    setFulfillableItems((state) => state.filter((i) => i.id !== itemId))
+    form.unregister(`quantity.${itemId}`)
+  }
+
+  const selectedLocationId = useWatch({
+    name: "location_id",
+    control: form.control,
+  })
+
+  return (
+    <RouteFocusModal.Form form={form}>
+      <form
+        onSubmit={handleSubmit}
+        className="flex h-full flex-col overflow-hidden"
+      >
+        <RouteFocusModal.Header>
+          <div className="flex items-center justify-end gap-x-2">
+            <RouteFocusModal.Close asChild>
+              <Button size="small" variant="secondary">
+                {t("actions.cancel")}
+              </Button>
+            </RouteFocusModal.Close>
+            <Button size="small" type="submit" isLoading={isMutating}>
+              {t("orders.fulfillment.create")}
+            </Button>
+          </div>
+        </RouteFocusModal.Header>
+        <RouteFocusModal.Body className="flex h-full w-full flex-col items-center divide-y overflow-y-auto">
+          <div className="flex size-full flex-col items-center overflow-auto p-16">
+            <div className="flex w-full max-w-[736px] flex-col justify-center px-2 pb-2">
+              <div className="flex flex-col divide-y">
+                <div className="flex-1">
+                  <Form.Field
```

Comment:
nit(non-blocking): Maybe we should default to the first item in `stock_locations`